polkadot/xcm/xcm-builder/src/currency_adapter.rs

```diff
@@ -16,6 +16,8 @@
 
 //! Adapters to work with `frame_support::traits::Currency` through XCM.
 
+#![allow(deprecated)]
```

Comment:
nit: seems that it's only a couple of places in code where we'd need `#[allow(deprecated)]`, so I'd prefer individual annotations instead of this , but it's fine either way.